test/routing/route_test.cc

```diff
@@ -0,0 +1,580 @@
+  EXPECT_EQ(kLaneAAPositionResult.distance, position_result.phase_position_result.distance);
+}
+
+TEST_F(RouteWithTwoPhasesTest, FindRoutePositionByInertialPositionIsMappedViaPhaseOntoLaneSRangeBA) {
+  // To simplify the modelled situation, please revisit the following diagram where the o
+  // indicates where the InertialPosition falls.
+  // <pre>
+  //
+  //      Phase A        Phase B
+  //  x------AA------*-o----BA------*
+  //  *------AB------x
+  //
+  // </pre>
+  const api::LanePositionResult kLaneAAPositionResult{api::LanePosition{100., 0., 0.},
+                                                      api::InertialPosition{100., 5., 0.}, 10.};
+  const api::LanePositionResult kLaneABPositionResult{api::LanePosition{100., 5., 0.},
+                                                      api::InertialPosition{100., 5., 0.}, 10.};
+  const api::LanePositionResult kLaneBAPositionResult{api::LanePosition{10., 0., 0.},
+                                                      api::InertialPosition{110., 5., 0.}, 0.};
+  const api::InertialPosition kInertialPosition{110., 5., 0.};
+  EXPECT_CALL(lane_a_a_, DoToLanePosition(_)).WillRepeatedly(Return(kLaneAAPositionResult));
+  EXPECT_CALL(lane_a_b_, DoToLanePosition(_)).WillRepeatedly(Return(kLaneABPositionResult));
+  EXPECT_CALL(lane_b_a_, DoToLanePosition(_)).WillRepeatedly(Return(kLaneBAPositionResult));
+  const Route dut({*phase_a_, *phase_b_}, road_network_.get());
+
+  const RoutePositionResult position_result = dut.FindRoutePosition(kInertialPosition);
+
+  EXPECT_EQ(kPhaseBIndex, position_result.phase_index);
+  EXPECT_EQ(kLaneSRangeBAIndex, position_result.phase_position_result.lane_s_range_index);
+  EXPECT_EQ(kLaneBAPositionResult.lane_position.s(), position_result.phase_position_result.lane_position.s());
+  EXPECT_EQ(kLaneBAPositionResult.lane_position.r(), position_result.phase_position_result.lane_position.r());
+  EXPECT_EQ(kLaneBAPositionResult.lane_position.h(), position_result.phase_position_result.lane_position.h());
+  EXPECT_EQ(kLaneBAPositionResult.nearest_position.x(), position_result.phase_position_result.inertial_position.x());
+  EXPECT_EQ(kLaneBAPositionResult.nearest_position.y(), position_result.phase_position_result.inertial_position.y());
+  EXPECT_EQ(kLaneBAPositionResult.nearest_position.z(), position_result.phase_position_result.inertial_position.z());
+  EXPECT_EQ(kLaneBAPositionResult.distance, position_result.phase_position_result.distance);
+}
+
+TEST_F(RouteWithTwoPhasesTest, FindRoutePositionByRoadPositionWhenRoadPositionIsNotInRouteBecomesByInertialPosition) {
+  // To simplify the modelled situation, please revisit the following diagram where the o
+  // indicates where the RoadPosition falls, and the m indicates the mapped position.
+  // <pre>
+  //
+  //      Phase A        Phase B
+  //  x------AA------*------BA------*
+  //  *------AB------m
+  //  s------AC------o
```

Comment:
Help me to understand this case. `AC` isn't part of the phase nor the route so when querying over that lane the returned value falls into `AB` due to the latter belonging to the phase. Is that correct?

Reply:
Correct, also I revisited the documentation.